### ISSUE TITLE
Navigate regions focus style bug with Safari

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#46111]
### Description

Follow-up to https://github.com/WordPress/gutenberg/pull/45369

Turns out the Navigate regions focus style isn't visible on the content area with Safari, under certain conditions. This seems to be a Safari bug, which is known to have problems with z-indexes.

The bug can be reproduced only when the post has some long content.

### Step-by-step reproduction instructions

**Test with Safari:**
- Edit a post with some short content, e.g. a couple one-line paragraphs.
- Press Control+backtick to navigate through the editor regions.
- Observe the focus style (a blue outline) is visible when focusing the content area, as expected. See the animated GIF below:

![short-content](https://user-images.githubusercontent.com/1682452/204246193-b100411f-3bd5-4663-ba02-18af3b7a23ed.gif)

- Edit a post with some long content that overflows the viewport.
- Press Control+backtick to navigate through the editor regions.
- Observe the focus style on the content area is not visible. See the animated GIF below:

![long-content](https://user-images.githubusercontent.com/1682452/204246518-59e699a2-e091-4a7d-8a47-dfab032fbd8c.gif)

- While the content area is focused, hover the mouse between two blocks to make the 'in between inserter' appear.
- Observe the focus style becomes visible when the in between inserter appears. See the animated GIF below:

![long-content-hover](https://user-images.githubusercontent.com/1682452/204247362-6b31dc23-941c-408d-ba26-0abedeff4ad3.gif)

### Environment info

- macOS Monterey 12.6
- Safari 16.0 (17614.1.25.9.10, 17614)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Can't reproduce any longer after recent changes. Closing.

Note: to make sure this kind of visual regressions don't happen again in the future, having some visual regression tests would be great. https://github.com/WordPress/gutenberg/pull/46210 is a proof of concept in that direction. However, introducing visual regression tests requires some higher-level discussion to establish a proper workflow.
